src/wallet/walletdb.cpp

```diff
@@ -819,21 +819,23 @@ void ThreadFlushWalletDB()
                 if (nRefCount == 0)
                 {
                     boost::this_thread::interruption_point();
-                    const std::string& strFile = pwalletMain->strWalletFile;
-                    map<string, int>::iterator _mi = bitdb.mapFileUseCount.find(strFile);
-                    if (_mi != bitdb.mapFileUseCount.end())
-                    {
-                        LogPrint("db", "Flushing %s\n", strFile);
-                        nLastFlushed = CWalletDB::GetUpdateCounter();
-                        int64_t nStart = GetTimeMillis();
-
-                        // Flush wallet file so it's self contained
-                        bitdb.CloseDb(strFile);
-                        bitdb.CheckpointLSN(strFile);
-
-                        bitdb.mapFileUseCount.erase(_mi++);
-                        LogPrint("db", "Flushed %s %dms\n", strFile, GetTimeMillis() - nStart);
+                    for (CWallet_ptr pwallet : vpwallets) {
```

Comment:
This thread is created from the non-static method `CWallet::postInitProcess`, so it seems like it makes more sense for it to continue flushing a single wallet, rather than going through the global wallets list. Instead of looping and accessing the global `vpwallet` variable, `ThreadFlushWalletDB` could just take a new `CWallet* pwallet` argument and be invoked with:

```
threadGroup.create_thread([this](){ThreadFlushWalletDB(this);});
```

from `CWallet::postInitProcess`.